### PR TITLE
added Spotify-Login button and deleted Greeting Controller

### PR DESCRIPTION
**Pull request by Lajmeno**

### CODE REVIEW

File: backend/src/main/java/de/neuefische/app/PLHeavenApplication.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+public class PLHeavenApplication {
+
+	public static void main(String[] args) {
+		SpringApplication.run(PLHeavenApplication.class, args);
+	}
+
+	@Bean
+	public RestTemplate restTemplate() {
+		return new RestTemplate();
+	}
```
Comment (Droggelbecher92) [inline backend/src/main/java/de/neuefische/app/PLHeavenApplication.java:18]: Do you really need the Bean in your main?

File: backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.RestTemplate;
+
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+import java.util.Base64;
+
+@RestController
+@RequestMapping("/callback")
```
Comment (Droggelbecher92) [inline backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java]:
```suggestion
@RequestMapping("/api/callback")
```

consider adding api or another prefix to your backend requests. With this prefix you'll have no conflicts with your Frontend-Router and can use the same url for the Frontend, without the prefix (frontend /users for userinfos, backend /api/users for userinfos)

File: backend/src/main/java/de/neuefische/app/spotify/SpotifyResponse.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package de.neuefische.app.spotify;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public record SpotifyResponse(
```
Comment (Droggelbecher92) [inline backend/src/main/java/de/neuefische/app/spotify/SpotifyResponse.java:5]: records 🥳

File: backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+                SpotifyResponse.class
+        );
+    }
+
+    HttpHeaders createGetTokenHeaders(){
+        return new HttpHeaders() {{
+            String forHeader = spotifyClientId+":"+spotifyAuthSecret;
+            String encodedString = Base64.getEncoder().encodeToString(forHeader.getBytes());
+            set( "Authorization", "Basic " + encodedString );
+            setContentType(MediaType.APPLICATION_FORM_URLENCODED);
```
Comment (Droggelbecher92) [inline backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java]:
```suggestion
HttpHeaders createGetTokenHeaders(){
    HttpHeaders header = new HttpHeaders();
        header.setBasicAuth(spotifyClientId,spotifyAuthSecret);
        header.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
        return header;
```
Spring can handle the encoding for you if you like. I think you'll also won't need setContentType

**Review by Droggelbecher92 — DISMISSED** [review]
This is fine 👍 Left a few comments, but works anyway :)

File: backend/src/main/java/de/neuefische/app/PLHeavenApplication.java (hunk 1 above)
Comment (Lajmeno) [reply backend/src/main/java/de/neuefische/app/PLHeavenApplication.java:18]: No but this way I can use it everywhere??

File: backend/src/main/java/de/neuefische/app/spotify/SpotifyResponse.java (hunk 3 above)
Comment (Lajmeno) [reply backend/src/main/java/de/neuefische/app/spotify/SpotifyResponse.java:5]: Haha just stealing from André :D

File: backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java (hunk 4 above)
Comment (Lajmeno) [reply backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java]: Ah cool! Good advise also

File: backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+HttpHeaders createGetTokenHeaders(){
+    HttpHeaders header = new HttpHeaders();
+        header.setBasicAuth(spotifyClientId,spotifyAuthSecret);
+        header.setContentType(MediaType.APPLICATION_FORM_URLENCODED);
+        return header;
+        }};
+    }
+
+
+}
```
Comment (Droggelbecher92) [inline backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java]:
```suggestion
```

My bad, should be ok now....

**Review by Droggelbecher92 — DISMISSED** [review]

File: backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java (hunk 5 above)
Comment (Lajmeno) [reply backend/src/main/java/de/neuefische/app/spotify/SpotifyApiController.java]: Haha yes your fault. I already corrected it locally :D

**Review by Droggelbecher92 — APPROVED** [review]
Nice!

**Pull request merged**